Screen texture and scan lines are now affected by curvature

CowThing opened the pull request:
The shader no longer multiplies what's behind it, and now uses the screen texture as the color, and applies curvature to the screen UV. Scanlines are also affected by curvature now. I also added an option to change the speed of the scanline movement.

![example](https://user-images.githubusercontent.com/11450875/61297457-47a84600-a7aa-11e9-8474-1c736b0eca5d.png)

hiulit commented:
Hey, thanks! That's awesome! :D
I've just opened an issue for myself to do what you have already did :P

The only issue I see is that the grille is also affected by the curvature.

<img width="752" alt="Captura de Pantalla 2019-07-16 a les 16 32 32" src="https://user-images.githubusercontent.com/481475/61303332-9f18d700-a7e7-11e9-9ce6-23f801d63606.png">

CowThing commented:
Yeah I noticed that as well, it's called a moire effect. It's caused by how the grille lines are being distorted by the curve. If you scale up the screen enough it goes away. This does happen on real CRT screens as well when you take a photograph of one. I don't know if it's possible to fix it.

hiulit commented:
That's a pity. The grille looked pretty cool, hehe!

It gets even worse when setting `viewport` mode.

<img width="752" alt="Captura de Pantalla 2019-07-16 a les 17 10 05" src="https://user-images.githubusercontent.com/481475/61307097-fd48b880-a7ed-11e9-993b-946887a6976f.png">

Even if you scale up the screen, the grille stays the same. The moire effect doesn't go away.

hiulit commented:
I'm really new to shaders, so here's my noob question: Can the grille somehow be "on top" of the rest?

CowThing commented:
Keep in mind this also happens when curvature is disabled. It's just a bit harder to tell when it happens because the moire effects are straight lines instead of curved lines. And yeah when it's set to viewport mode it wont go away when scaled up. This effect can happen to the scan lines as well, not just the grille lines.

I made a few changes to help. I changed the line calculations from using `cos` to `sin`, and increased the precision of PI, so that the lines better match the pixels. I also added a `screen_size` uniform which will allow the user to change how many grille lines and scan lines appear, having fewer grille lines and scan lines will make them larger, which makes it harder for the moire effect to appear.

hiulit commented:
Thank you very much! I really appreciate it.
